From 419cac8590c4be4f6b4b45ea7afb91e5443c4075 Mon Sep 17 00:00:00 2001
From: yuyue <yuyue@mprc.pku.edu.cn>
Date: Fri, 4 Nov 2011 15:25:16 +0800
Subject: [PATCH 073/641] UniCore64: Add function __switch_to

Fixed error:
elfcore.c:(.sched.text+0x888): undefined reference to `__switch_to'

Signed-off-by: yuyue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/Makefile |    2 +-
 arch/unicore64/kernel/entry.S  |    8 ++++++++
 2 files changed, 9 insertions(+), 1 deletion(-)
 create mode 100644 arch/unicore64/kernel/entry.S

diff --git a/arch/unicore64/kernel/Makefile b/arch/unicore64/kernel/Makefile
index abfba64..239ef31 100644
--- a/arch/unicore64/kernel/Makefile
+++ b/arch/unicore64/kernel/Makefile
@@ -1,4 +1,4 @@
 obj-y			:= ptrace.o setup.o init_task.o process.o
-obj-y			+= time.o traps.o irq.o cpu-ucv3.o sys.o
+obj-y			+= time.o traps.o irq.o cpu-ucv3.o sys.o entry.o
 head-y			:= head.o
 extra-y			:= $(head-y) vmlinux.lds
diff --git a/arch/unicore64/kernel/entry.S b/arch/unicore64/kernel/entry.S
new file mode 100644
index 0000000..b5bbd6e
--- /dev/null
+++ b/arch/unicore64/kernel/entry.S
@@ -0,0 +1,8 @@
+#include <linux/linkage.h>
+
+/**
+ * __switch_to - Register switch for UniCore3 processors
+ */
+ENTRY(__switch_to)
+	/* FIXME: */
+ENDPROC(__switch_to)
-- 
1.7.9.5

